features/FxA-16-devices-view/README.md

```diff
@@ -4,5 +4,174 @@ Device View - Basic UI
+Stories
+-------
 
+As a FF user when using Firefox,
+I want to know the devices that are connected to my Firefox Account,
+and important information about the status of those devices.
+
+I want my FF devices to be logically named
+so I recognize them and have the ability to rename them.
+I want to easily see the type and browser details of the each device,
+and to be able to identify the device I am currently using.
+
+I want a place to go when I think that something might be wrong with Sync,
+to see when each device last checked in with the Mozilla servers.
+
+I want a place to go when I think that my account
+may have been used without my permission,
+to check what devices are connected
+and to disconnect devices that I do not recognize.
+
+
+UI Mocks
+--------
+
+@ryanfeeley's mocks go here
+
+
+Details
+-------
+
+#### What is a "device"?
+
+By "device" here we mean "an instance of Firefox".
+There is no separate notion of having multiple "browsers" on a single "device".
+Technically the concept is closer to "profiles"
+but this is not a good name to present the users.
+
+* When I use a desktop machine
+  and connect both Firefox Release and Firefox Nightly to my account,
+  they appear as two separate devices.
+  There will be no indicatiion that they're on the same physical machine
+  apart from similarity in their auto-generated name and their UA details.
+
+* When I use Firefox for Android or Firefox for iOS...
+  TODO what exactly do we expect to happen?
+
+* When I use a FirefoxOS device,
+  it appears in the view as a single device
+  since the connetion is managed at the system level.
+  Also, TODO do we care at this stage?
+
+For the initial version, only things that connect to sync count as devices.
+
+* When I log in to Marketplace with my Firefox Account,
+  using the web-based flow in a browser that is not connected to my account,
+  the resulting session does *not* appear in the list of devices.
+
+* When I log in to Pocket with my Firefox Account
+  using non-Firefox web browser,
+  the resulting session does *not* appear in the list of devices.
+
+* When I have existing Firefox instances
+  that are already connected to sync,
+  they will appear in the devices view automatically.
+
+* TODO this also means that FirefoxOS should not be in the list
+  because it doesn't connect to sync yet.
+  It will soon though and it's something for us to think about.
+  
+
+
+#### What details do we show about the device?
+
+For each device we will show:
+
+* The human-readable name for the device,
+  either an auto-generated one or one provided by the user.
+
+* The time it was last seen,
+  i.e. the times at which it last used its session token
+  to interact with the FxA servers.
```

Comment:
Nicely spotted, this is indeed a thorny issue for us, see also my comment at https://github.com/mozilla/fxa/issues/62#issuecomment-144864586.  I hadn't really thought about the "up to a 12 hour difference" part though.

The difficulty is that we don't currently track the last-synced-time anywhere, so we have to either descope the requirements, or expand the engineering work.  Let's use this PR as an opportunity to all agree explicitly on the best path forward.

I see two options technically:

1. Use the time of the last call to `/cert/sign` on the fxa servers, which we track already.
  * Pros: significantly simpler to implement for initial version
  * Cons: may be up to 12 hours stale due to client caching
2. Figure out how to track last-synced time on a per-device basis from server data
  * Pros: more useful and timely information for the user
  * Cons: surprisingly tricky given our current architecture (although I have some ideas)

These correspond to the following usre experiences.

 As a Firefox User, I explicitly "sync now" and then go do my devices view, and I see that the current device was last connected at:

1. A coarse-grained timestamp like "Today"
2. A fine-grained timestamp like "a few moments ago"

Do we expand our definition of "minimal" to encompass the work needed for (2), or do we contract the scope of "viable" so that we can initially ship (1)?